### PR TITLE
🐛 Metrics: ensure instrumentation messages are treated only once

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
Since @mrnicegyu11 fixed how prometheus scrapes instances of the webserver, the metrics coming out of it were multiplied by the number of instances. The bugfix was correct, but then the metrics were wrong.
The final issue was identified as being a flaw in how the webserver gets the metrics out of RabbitMQ.
It was using ```exclusive``` queues (1 per instance of the webserver), thus replicating the metrics in all instances.
Now it uses a non-exclusive queue to this end.

## Related issue/s
fixes https://github.com/ITISFoundation/osparc-simcore/issues/3472

### CODE REVIEW

**Review by mrnicegyu11 — APPROVED** [review]
good catch, thanks a lot!

**Review by elisabettai — APPROVED** [review]
Thanks a lot for this and the super-quick fix!

**Review by pcrespov — APPROVED** [review]

File: packages/service-library/src/servicelib/rabbitmq.py [hunk 1]
```diff
@@ -110,11 +117,15 @@ async def subscribe(
...
-            queue = await channel.declare_queue(
+            queue_parameters = dict(
                 durable=True,
-                exclusive=True,
+                exclusive=exclusive_queue,
                 arguments={"x-message-ttl": _RABBIT_QUEUE_MESSAGE_DEFAULT_TTL_S},
             )
+            if not exclusive_queue:
+                # NOTE: setting a name will ensure multiple instance will take their data here
+                queue_parameters |= {"name": exchange_name}
```
Comment (GitHK) [inline packages/service-library/src/servicelib/rabbitmq.py:127]: I think above is way too fancy for one single element.
```suggestion
                queue_parameters["name"] = exchange_name
```

**Review by GitHK — APPROVED** [review]
👍 nice

File: packages/service-library/src/servicelib/rabbitmq.py (hunk 1 above)
Comment (sanderegg) [reply packages/service-library/src/servicelib/rabbitmq.py:127]: I like fancy. and it's only the beginning...

**Pull request merged**